apps/server/src/shared/testing/factory/base.factory.ts

```diff
@@ -49,6 +50,19 @@ export class BaseFactory<T, U, I = any> {
 		return entity;
 	}
 
+	/**
+	 * Build an entity using your factory and generate a id for it.
+	 * @param params
+	 * @param id
+	 * @returns an entity
+	 */
+	buildWithId(params: DeepPartial<U> = {}, id?: string): T {
```

Comment:
do we really need that? we have to discuss...

Reply:
Yes i think so ..otherwise we must add this to nearly and unit test that work with id. We have a lot of unit test that test at the moment to element.id (=null) === null. The test show green in situation when they shoul fail or fail when they should work. If you look outside of the task context the most remove factorys and and do not use it for test because it to not work like expected.

Reply:
sorry, i can't hardly find any tests where setting the id is **really** necessary. maybe we should look at this together. also the `buildWithId` seems a little bit inconsistent to me because it doesn't apply to lists and doesn't work recursively. (please don not add a `buildListWithIds` :-) )

Reply:
Every test that create an status for task works with id matches. You can not really say that it is fine for you that we pass user.id = null to create status and execute in it a [null].includes(null) in test cases. 
If we add a check that we only allow a valid id as input all the task unit test will fail. If you have different users in the list you have [null, null, null, ..].includes(null). 
You can pass student user.id=null and if a substitute user with null exist, it will match for isSubstitute and set it to true, we have in entity unit test different implementations for hacking ids in it, or on other places where the factorys removed from unit test because they do not work like expected (in context for null) and so on...

For me it is total fine to have no list operations for it.

Reply:
Discussion is outsourced to meeting